### PR TITLE
feat: add `$.f64`

### PR DESCRIPTION
**Pull request by tjjfvi**

### CODE REVIEW

**Comment by harrysolovay** [comment]
Tests failing

File: codecs/float.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const f64 = createCodec<number>({
+  _metadata: metadata("$.f64"),
+  _staticSize: 8,
+  _encode(buffer, value) {
+    buffer.view.setFloat64(buffer.index, value, true)
+    buffer.index += 8
+  },
+  _decode(buffer) {
+    const value = buffer.view.getFloat64(buffer.index, true)
+    buffer.index += 4
```
Comment (ryanleecode) [inline codecs/float.ts:12]: How come when we are encoding we add +8 but decoding its +4?

Comment (tjjfvi) [reply codecs/float.ts]: Thanks, was a typo

**Comment by tjjfvi** [comment]
> Tests failing –– could this be the result of the index increment in `_decode` (4, not 8)?

This bug was actually going uncaught by the tests, as I hadn't tested anything like `$.array($.f64)`, so it didn't care about the index. I've updated it so that `testCodec` checks that the full buffer is consumed.

The tests were actually failing because I failed to remove some old snapshots.

Should all be good now.

**Review by ryanleecode — APPROVED** [review]

**Pull request merged**